tracing/src/main/java/com/palantir/tracing/CloseableTracer.java

```diff
@@ -39,6 +40,13 @@ public static CloseableTracer startSpan(String operation) {
         return startSpan(operation, SpanType.LOCAL);
     }
 
+    /**
+     * Opens a new {@link SpanType#LOCAL LOCAL} span for this thread's call trace, labeled with the provided operation.
+     */
+    public static CloseableTracer startSpan(Supplier<String> operation) {
```

Comment:
I'm worried that clients might choose to use this instead of the normal one for plain string concatenations which are not that expensive, thereby potentially slowing down hot paths.

Reply:
Today hot paths are slow depending on the length of the http path.